libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,64 @@
+// (C) 2020 GoodData Corporation
+import React, { useEffect, useState, useRef } from "react";
+import { useBackend, useWorkspace } from "@gooddata/sdk-ui";
+import { InvariantError } from "ts-invariant";
+import { ITheme } from "@gooddata/sdk-model";
+import { IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
+
+import { setCssProperties } from "../cssProperties";
+import { ThemeContextProvider } from "./Context";
+
+/**
+ * @beta
+ */
+export interface IThemeProviderProps {
+    backend?: IAnalyticalBackend;
+    workspace?: string;
+    loader?: string;
+}
+
+/**
+ * @beta
+ */
+export const ThemeProvider: React.FC<IThemeProviderProps> = ({
+    children,
+    backend: backendParam,
+    workspace: workspaceParam,
+}) => {
+    const backend = backendParam || useBackend();
```

Comment:
Thinking about this, this will probably break hooks as well. If you for example first specify backend and then change it so that it is no longer specified, there will be more hook calls than before which will blow up iinm.